enterprise/api/filters.py

```diff
@@ -128,17 +127,23 @@ def filter_queryset(self, request, queryset, view):
 
 class EnterpriseCustomerInviteKeyFilterBackend(filters.BaseFilterBackend):
     """
-    Filter backend to return invite keys under the user's enterprise only.
+    Filter backend to return invite keys under the user's enterprise(s) only.
+    Supports filtering by enterprise_customer_uuid.
 
     * Staff users will bypass this filter.
     """
 
     def filter_queryset(self, request, queryset, view):
-        """
-        Filter out keys that are not part of the user's enterprise.
-        """
+        query_params = {}
+
         if not request.user.is_staff:
-            enterprise_customer_uuid = get_enterprise_customer_from_user_id(request.user.id)
-            queryset = queryset.filter(enterprise_customer_id=enterprise_customer_uuid)
+            user_enterprise_customer_uuids = [
+                ec.enterprise_customer_id for ec in EnterpriseCustomerUser.objects.filter(user_id=request.user.id)
+            ]
+            query_params.update(enterprise_customer_id__in=user_enterprise_customer_uuids)
 
-        return queryset
+        enterprise_customer_uuid = request.query_params.get('enterprise_customer_uuid', None)
```

Comment:
Since this is on the request object I'm assuming it's user input. Is it safe for us to use this to query this info? Could an actor provide an enterprise_customer_uuid that wasn't theirs and gain access to info they shouldn't have?

Would it be better to get the info enterprise_customer info from the request.user object?

Reply:
If the user is not staff, they can only view keys in enterprises they are part of.

Reply:
Fair enough, I didn't put that bit together.